packages/react-next/src/components/Checkbox/examples/Checkbox.Other.Example.tsx

```diff
@@ -26,11 +26,32 @@ export const CheckboxOtherExample: React.FunctionComponent = () => {
 
       <Checkbox label="Checkbox with extra props for the input" inputProps={inputProps} />
 
-      <Checkbox label="Checkbox with link inside the label" onRenderLabel={_renderLabelWithLink} />
+      <Checkbox label={<span>Custom label 1</span>} />
+
+      <Checkbox label={{ className: 'label', children: 'Custom label 2' }} />
+
+      <Checkbox label={{ children: CustomLabel }} />
+
+      <Checkbox onRenderLabel={_renderLabelWithLink} />
     </Stack>
   );
 };
 
+const CustomLabel: React.FunctionComponent<{
```

Comment:
`children` are render function, they are not React component and should not be them as they implement render props pattern:

https://reactjs.org/docs/render-props.html